components/new-contribution-flow/StepProfileLoggedInForm.js

```diff
@@ -27,6 +27,9 @@ const NewContributionFlowStepProfileLoggedInForm = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [defaultSelectedProfile]);
 
+  // if admin of collective you are donating to, remove it from the list
+  remove(profiles, p => p.id === collective.legacyId);
```

Comment:
It's better to not mutate props. I missed that we were already doing it in the previous component, but we should review that and have a `const filteredProfiles = React.useMemo(() => filterProfiles(profiles), [profiles])` instead.

Two reasons for that:
- When components start to mutate their props it's hard to keep track of where the changes are coming from.
- In its latest version, Apollo returns only immutable results, which means doing something like `data.profiles.push` will actually crash